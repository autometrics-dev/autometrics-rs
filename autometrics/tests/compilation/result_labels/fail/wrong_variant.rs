// This test ensures that the macro fails with a readable error when the
// attribute given to one variant inside the enumeration does not use one of the
// predetermined values (that would make the automatic queries fail, so the
// macros need to forbid wrong usage at compile time)
use autometrics_macros::ResultLabels;

struct Inner {}

#[derive(ResultLabels)]
enum MyError {
    Empty,
    #[label(result = "not ok")]
    ClientError {
        inner: Inner,
    },
    ServerError(u64),
}

fn main() {}
