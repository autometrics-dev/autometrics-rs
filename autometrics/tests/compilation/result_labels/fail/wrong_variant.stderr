error: Only "ok" or "error" are accepted as result values
  --> tests/compilation/result_labels/fail/wrong_variant.rs:12:22
   |
12 |     #[label(result = "not ok")]
   |                      ^^^^^^^^
